src/core/download_list.cc

```diff
@@ -606,8 +606,8 @@ DownloadList::received_finished(Download* download) {
   if (rpc::call_command_value("pieces.hash.on_completion"))
     // Set some 'checking_finished_thingie' variable to make hash_done
     // trigger correctly, also so it can bork on missing data.
-    hash_queue(download, Download::variable_hashing_last);
-    confirm_finished(download);
+    if (hash_queue(download, Download::variable_hashing_last))
+      confirm_finished(download);
```

Comment:
```
  if (rpc::call_command_value("pieces.hash.on_completion"))
    // Set some 'checking_finished_thingie' variable to make hash_done
    // trigger correctly, also so it can bork on missing data.
    if (hash_queue(download, Download::variable_hashing_last))
      confirm_finished(download);
  else
    confirm_finished(download);
```

The above code was exactly how I intended it to be.

With braces:

```
if (rpc::call_command_value("pieces.hash.on_completion"))
{
    // Set some 'checking_finished_thingie' variable to make hash_done
    // trigger correctly, also so it can bork on missing data.
    if (hash_queue(download, Download::variable_hashing_last)) // if this returns true then we confirm finished, else we don't
    { confirm_finished(download); }
}
else // if pieces.hash.on_completion is disabled
{ confirm_finished(download); }

Admittedly, I'm not a programmer, so if this is flawed code, hopefully the intent is conveyed in the comments I have added and someone with better skill can fix it.